### ISSUE TITLE
The sms code has expired. Please re-send the verification code to try again

### ISSUE THREAD

**Issue by Abouassi98** [firebase/flutterfire#11251]
## Bug report

**Describe the bug**
The sms code has expired. Please re-send the verification code to try again

### Steps to reproduce

When I send code from my real device, I encounter this bug before 120 seconds elapse.but when I resend code this message don't show

### Expected behavior

don't show this error message before detected time as a 120 seconds

### Sample project

```
FirebaseAuth.instance.verifyPhoneNumber(
          phoneNumber: phone,
          timeout: const Duration(seconds: 60),
          verificationCompleted: _verificationCompleted,
          verificationFailed: _verificationFailed,
          codeSent: codeSent,
          codeAutoRetrievalTimeout: (verId) {
            verificationId = verId;
          },
        );


Future<void> _verificationCompleted(PhoneAuthCredential credential) async {
    try {
      await FirebaseAuth.instance.currentUser!.reload();
      if (FirebaseAuth.instance.currentUser == null) {
        await FirebaseAuth.instance.signInWithCredential(credential);
      } else {
        emit(SendCodeErrorState('هنا خطا ما'));
      }
    } on FirebaseAuthException catch (e) {
      emit(SendCodeErrorState(e.toString()));
    }
  }

  void codeSent(String verifId, int? c) {
    verificationId = verifId;
    emit(SendCodeSuccessState('sucess'));
    log('Code Sent');
    return null;
  }

```
---

## Additional context

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.5, on macOS 13.4.1 22F82 darwin-x64, locale en-EG)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] VS Code (version 1.78.2)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
- firebase_auth 4.6.3 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 2.14.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_crashlytics 3.3.3 [firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface flutter stack_trace]
- firebase_messaging 14.6.4 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
```

</details>

---

**Comment by aleksandar-radivojevic** [comment]
I also have this issue.

**Comment by AbdullahGaber** [comment]
Same issue here

**Comment by sleem2012** [comment]
Same issue

**Comment by darshankawar** [comment]
Thanks for the report @Abouassi98 
Looking at the error received, it seems that the SMS code is verified as soon as it comes on the phone, and the user is created. Can you try to set the timeout to 5 seconds and see if it still causes the error ?

**Comment by Abouassi98** [comment]
I have tried that solution, but it does not fix this issue. I have also attempted to pass an empty function to `verificationCompleted`, but nothing changes.

**Comment by darshankawar** [comment]
Can you try the [plugin example](https://github.com/firebase/flutterfire/tree/master/packages/firebase_auth/firebase_auth/example) and check if using it, you get same behavior or not ?

**Comment by Abouassi98** [comment]
https://github.com/firebase/flutterfire/assets/53384802/f254c5ef-9397-454b-b422-f8d7d9bd453d

 I have faced this issue

**Comment by darshankawar** [comment]
@Abouassi98 Thanks for the update.
When the error is received, have you checked in the console if the user gets added in the firebase or not ?
Also, can you try to put a listener to auth changes on login screen and see if it helps ?
Moreover, just to confirm that the issue isn't device / OS specific, can you try the same on other devices and see if you still get the same error or not ?

**Comment by Abouassi98** [comment]
Yeah, the phone number has been added to Firebase. However, this issue only occurs in a specific case: when sending a message in release mode and when the client sends a code from a phone with a SIM card. So, when I try to put a listener on the login screen to detect authentication changes, my phone number gets blocked because I have tried more than three times.

Additionally, I have tried this on three other phones and encountered the same issue.
